### PR TITLE
[TTAHUB-3627] Add citation alerts

### PR DESCRIPTION
**Pull request by AdamAdHocTeam**
## Description of change

**NOTE:** Before testing move the citations start date back in the src\services\citaion.js to ensure you get some citations to populate in the drop down.

The primary focus of this PR is to add an alert and error, for grants missing citations when the only goal on the report is a monitoring goal.

1. The warning shows on the goals and objectives page right above the citations drop down and should list any recipients that fit this criterica.
2. The error on the review and submit page and also shows the list of recipients missing citations and prevents submitting until its fixed.

In addition this fixes some other issues I found:

1. Weird case if you had two grants with citations but only saved a citation for one on the objective (server side error that would fail to save any citations).
2. Citations drop down and warning would suddenly popup on the objective form (switching to the goal and objectives from another page like summary).

## How to test

- Review the code.
- Make sure the alert shows on the goals and objective page.
- Make sure the error shows on the review and submit page and prevents submission.
- Ensure the warning/error goes away once the citation is added or grant removed.

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-3627

### Every PR

- [x] Meets issue criteria
- [x] JIRA ticket status updated
- [x] Code is meaningfully tested
- [x] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete
- [ ] QA review complete

### Before merge to main

- [ ] OHS demo complete
- [ ] Ready to create production PR

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [x] Update JIRA ticket status

### CODE REVIEW

File: frontend/src/pages/ActivityReport/Pages/Review/Submitter/Draft.js [hunk 1]
```diff
@@ -77,7 +78,10 @@ const Draft = ({
   };
 
   const onSubmit = (e) => {
-    if (allGoalsHavePromptResponses && !hasIncompletePages && !grantsMissingMonitoring.length) {
+    if (allGoalsHavePromptResponses
+      && !hasIncompletePages
+      && !grantsMissingMonitoring.length
+      && !grantsMissingCitations.length) {
```
Comment (thewatermethod) [inline frontend/src/pages/ActivityReport/Pages/Review/Submitter/Draft.js:84]: might be nice to move this into a variable outside of this function, like 

```js
const canSubmitReport = allGoalsHavePromptResponses
      && !hasIncompletePages
      && !grantsMissingMonitoring.length
      && !grantsMissingCitations.length;
      
const onSubmit = (e) => {
    if (canSubmitReport) {

Comment (thewatermethod) [reply frontend/src/pages/ActivityReport/Pages/Review/Submitter/Draft.js:84]: (totally optional)

File: frontend/src/pages/ActivityReport/Pages/Review/Submitter/index.js [hunk 2]
```diff
@@ -105,7 +105,49 @@ const Submitter = ({
...
+          monitoringReferenceGrantIds.forEach((grantId) => {
+            if (!acc.has(grantId)) {
+              acc.add(grantId);
+            }
+          });
+          return acc;
+        }, new Set(),
+      ));
+
+      // console.log('hasMonitoringGoalSelected', hasMonitoringGoalSelected);
```
Comment (thewatermethod) [inline frontend/src/pages/ActivityReport/Pages/Review/Submitter/index.js:143]: remove console

File: frontend/src/pages/ActivityReport/Pages/components/Objective.js [hunk 3]
```diff
@@ -406,7 +441,41 @@ export default function Objective({
...
+                    {' '}
+                    <Link to={`/activity-reports/${reportId}/activity-summary`}>Activity summary</Link>
+                  </li>
+                  <li>
+                    Add another goal to the report
+                  </li>
+                </ul>
+              </span>
+            </span>
+          </Alert>
```
Comment (thewatermethod) [inline frontend/src/pages/ActivityReport/Pages/components/Objective.js:474]: are these alerts identical except for the type? Might be a good idea to turn these into a component

**Review by thewatermethod — APPROVED** [review]
Seems to be firing at the correct time; might want to run the implementation by the UI team as it isn't really firing on blur but on change

**Comment by AdamAdHocTeam** [comment]
> Seems to be firing at the correct time; might want to run the implementation by the UI team as it isn't really firing on blur but on change

It should be on change, on blur was just a suggestion confirmed with Lauren.

File: frontend/src/pages/ActivityReport/Pages/components/Objective.js (hunk 3 above)
Comment (AdamAdHocTeam) [reply frontend/src/pages/ActivityReport/Pages/components/Objective.js:474]: I think there are enough differences that the component will be simply and alert.

File: frontend/src/pages/ActivityReport/Pages/Review/Submitter/index.js (hunk 2 above)
Comment (AdamAdHocTeam) [reply frontend/src/pages/ActivityReport/Pages/Review/Submitter/index.js:143]: revmoed.

File: frontend/src/pages/ActivityReport/Pages/Review/Submitter/Draft.js (hunk 1 above)
Comment (AdamAdHocTeam) [reply frontend/src/pages/ActivityReport/Pages/Review/Submitter/Draft.js:84]: updated.

**Pull request merged**